### PR TITLE
Add fixture library for e2e tests

### LINKED ISSUE #6914: E2EFix: Create comprehensive fixture library (GraphQL and UI)
## Summary
Establish a structured fixture library for deterministic tests with realistic GraphQL responses and UI samples.

## Why
- Current fixtures are minimal and inconsistent
- Curated fixtures speed authoring and reduce duplication

## Scope
- Create cypress/fixtures/auth, admin, user, api/graphql directories
- Define naming: {domain}/{entity}-{scenario}.json or api/graphql/{operation}.{scenario}.json
- Seed datasets with ids, pagination cursors, timestamps
- Document usage with GraphQL utilities in cypress/README.md

## Implementation notes
- Keep datasets minimal yet representative; include edge entries
- Use i18n-safe text; avoid PII

## Starter Code
This sample code is only representative of what could be done. It is provided only to give a general outline. You will need to use your initiative and industry best practices to find a solution that is suitable for the application.
```ts
// cypress/fixtures/api/graphql/organizations.success.json
{
  "data": {
    "organizations": [
      { "id": "org_001", "name": "Example Org", "description": "Primary for tests", "createdAt": "2025-12-01T00:00:00.000Z" },
      { "id": "org_002", "name": "Long Name Organization — مرحبا 世界", "description": "Edge case", "createdAt": "2025-12-02T00:00:00.000Z" }
    ]
  }
}
```

## Files to Edit/Create
Due to recent refactoring, the locations of some screens and components may have changed to new sub-directories

### Files to Edit
- cypress/README.md

## 
## Acceptance criteria
- Fixture directories exist with representative success and error responses for major operations
- Example usage documented in cypress/README.md and verified in at least one example spec

## Out of scope
- Exhaustive coverage of entire GraphQL schema initially

## Blocks
- #6920
- #6922
- #6923
- #6924
- #6929
- #6935

## Blocked by
- None

## Labels
- refactor
- test

## Estimate
- 16 hours (1 PR)

## References
- https://docs-admin.talawa.io/docs/developer-resources/e2e-testing
- https://docs-admin.talawa.io/docs/developer-resources/reusable-components

### PR DESCRIPTION
**Pull request by Ansingh0305**
**What kind of change does this PR introduce?**

Refactor / test infrastructure (fixture library + documentation updates)

**Issue Number:**

Fixes #6914

**Snapshots/Videos:**

image

**If relevant, did you update the documentation?**

Yes — updated `cypress/README.md` and `docs/docs/docs/developer-resources/e2e-testing.md`.

**Summary**

Creates a structured Cypress fixture library (auth/admin/user/api-graphql) with minimal, representative datasets and edge cases. Adds GraphQL response fixtures for key operations and documents how to use them with GraphQL utilities to keep E2E tests deterministic and reusable.

**Does this PR introduce a breaking change?**

No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Other information**

Fixtures are intentionally minimal, i18n‑safe, and include edge cases (empty arrays, long names, Unicode) to improve test robustness without PII.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

* **Documentation**
  * Added a Fixture Library note pointing to guidance on fixture structure, GraphQL mocking examples, mock cleanup, and test isolation.

* **Tests**
  * Added numerous new test data fixtures (admin, user, auth, API), removed an old advertisement fixture.
  * Updated tests to load credentials from the new fixtures and to clear GraphQL mocks after each test.

* **Tests / Utilities**
  * Added a global GraphQL mock cleanup command, optional mock response-delay support, and clearer login error messaging.

### CODE REVIEW

**Comment by Ansingh0305** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where mock cleanup is important and response times are highly variable. Also please check if the best practices have been followed and production level code has been written. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Comment by Ansingh0305** [comment]
@coderabbitai please check if all the issues that you stated before have been solved and also you might not be having the access to the documentation md files hence you cannot detect but rest assured they are there. Please check if the other issues have been solved and this pr is ready for merge

**Comment by Ansingh0305** [comment]
@palisadoes @Cioppolo14 PTAL i have addressed the coderabbit comments.

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where mock cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

File: cypress/README.md [hunk 1]
```diff
@@ -4,3 +4,90 @@ Documentation on how to run cypress end to end testing can be found either:
 
 1. [Online](https://docs-admin.talawa.io/docs/developer-resources/e2e-testing) or,
 2. [Locally](../docs/docs/docs/developer-resources/e2e-testing.md)
+
+## Fixture Library
+
+Fixtures live under `cypress/fixtures/` and are organized by domain for
+consistency and reuse:
```
Comment (palisadoes) [inline cypress/README.md]: Move these edits to:
```
docs/docs/docs/developer-resources/e2e-testing.md
```

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Move the `cypress/README.md` edits to:

    ```
    docs/docs/docs/developer-resources/e2e-testing.md
    ```
1. Address the testing flakiness issues found here:
    - https://github.com/PalisadoesFoundation/talawa-admin/pull/7012#issuecomment-3845396344

**Comment by palisadoes** [comment]
@coderabbitai

Update your learning for all repositories under https://github.com/PalisadoesFoundation to evaluate the presence or absence of files based on the directory tree of the target branch of the pull request, not the default branch

**Comment by Ansingh0305** [comment]
@palisadoes I believe we should make a seperate issue for the mock isolation and repo wide adoption of the cy.ts files since this is out of scope for this issue and would require changing unrelated files. Would you like me to create the issues ?

**Comment by Ansingh0305** [comment]
> @Ansingh0305
> 
> All the content in this file should be moved to the markdown in `docs/docs/docs/developer-resources/e2e-testing.md`
> 
> If that is done, then there shouldn't be a need to have this done in two locations. It creates incomplete documentation.
> 
> If it means there needs to be two sections during the transition, then they should both be in `docs/docs/docs/developer-resources/e2e-testing.md`

@palisadoes 
Thank you for the explanation. I have moved the content to e2e-testing.md and have reduced README.md to a pointer. 
Please check if this is fine or if any more changes are required.

**Comment by Ansingh0305** [comment]
@palisadoes PTAL
